### PR TITLE
Create `yotpo_rich_snippets` table if absent during module upgrade

### LINKED ISSUE #144: yotpo_rich_snippets table was not created when upgrade from previous version.
Hi Yotpo,

The `yotpo_rich_snippets` table was not created when update from lower version, maybe you could check and create the table if the table do not exists.

### PR DESCRIPTION
**Pull request by fredden**
Fixes: #144

From what I can tell this table was introduced in version 2.4 in an install schema script (see 02b6571e), but the upgrade schema was never updated to match. This means that if a version prior to v2.4 was installed, the required table is absent. Upgrading to version 3.1.0 or later produces an error due to 2f2a3a2c assuming that the table exists. This pull request resolves that problem.

### CODE REVIEW

**Review by pniel-cohen — CHANGES_REQUESTED** [review]
That's fine excepts that this table should be installed using the default connection & not the `$salesConnection`.

**Review by pniel-cohen — COMMENTED** [review]
Just by looking, it seems fine to me.
*Still requires relevant QA before merging!

File: Setup/UpgradeSchema.php [hunk 1]
```diff
@@ -220,10 +220,53 @@ public function upgrade(SchemaSetupInterface $setup, ModuleContextInterface $con
             }
         }
 
-        $richSnippetsTable = $installer->getConnection()->describeTable($installer->getTable('yotpo_rich_snippets'));
+        if (!$defaultConnection->isTableExists($yotpoRichSnippetsTable)) {
```
Comment (sharp80) [inline Setup/UpgradeSchema.php:223]: we already have the code of creating this table - we need to make sure we don't duplicate this code.

Comment (fredden) [reply Setup/UpgradeSchema.php:223]: Hi sharp80. Did you get a chance to read through the pull request description and linked issue? I don't think I understand your comment here. There is code to create the table upon module installation, but not on upgrade. Are you asking for a code change?

Comment (sharp80) [reply Setup/UpgradeSchema.php:223]: @fredden - i meant that we have the same code that creates richSnippets table in a different file (InstallScheme.php) and better practice is to move the actual code that creates the table to utility function and call it from the upgradeScheme and from installScheme.

Comment (pniel-cohen) [reply Setup/UpgradeSchema.php:223]: What @fredden meant was:
The current installation code only runs on installation & not on upgrade.
On an extremely rare situation, when someone had a very old version of the yotpo module installed & now upgrading to a newer version, the table won't be created.
So it's not a bad idea to put that on the upgrade script as well, just i case.
Also, I think we should migrate to declarable scheme anyway soon, so it won't matter.

Comment (fredden) [reply Setup/UpgradeSchema.php:223]: Thanks for explaining. I've removed the duplicate code.

**Review by lgreenboim — APPROVED** [review]

**Comment by lgreenboim** [comment]
Thank you fo the fix, it’ll be included in the next release - 3.1.3

**Pull request closed without merging**